client/src/components/AddFieldAgent.js

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+
+function AddFieldAgent({ handleAdd, handleCancel }) {
+  const [firstName, setFirstName] = useState('');
```

Comment:
I personally like to track a single object, especially as models grow more complicated. But these controlled components are really nice. Well done.